### DISCUSSION TITLE
Is there a function for the webclient to do autoplay?

### DISCUSSION THREAD

**Discussion by Platypuschan** [snapcast/snapcast#1081] [Q&A]
Is there a way to get a webclient to autoplay? So the client would start playing the stream as soon as the webpage gets loaded..

**Comment by kingosticks** [comment]
Didn't web browsers do lots of work to explicitly remove that functionality to combat all those stupid video ads abusing it?

**Reply by jerblack** [reply]
Yeah, the browser window needs focus to play audio. For example, if you install a Chrome extension that lets you run your own JS on a page (like this one: https://chromewebstore.google.com/detail/user-javascript-and-css/nbhcbdghjpllgmfilhnhkllmkecfmpld?pli=1) and add this to your Snapweb page with the extension:
```js
setTimeout(()=>{document.getElementById("play-button").click()}, 2000)
```
when you refresh, after a couple seconds the play button gets clicked but no audio comes, and the console logs `snapstream.js:446 Failed to get chunk, read: 0/3528, chunks left: 0`. If you refresh the page and then immediately click the page before the setTimeout fires, autoplay begins properly as soon as it fires.

You can change this behavior per-site though in Chrome. Click the "Not secure" button on the left end of the address bar and go to site settings, and then in there set Sound to "Allow". Now, when you refresh the SnapWeb page with the JS above in place audio will begin playing 2 seconds after the page loads.
